### PR TITLE
integrate editor n plots2

### PR DESCRIPTION
**Pull request by keshav234156**
Fixes https://github.com/publiclab/PublicLab.Editor/issues/365

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by keshav234156** [comment]
@jywarren @VladimirMikulic the instructions mentioned in https://github.com/publiclab/PublicLab.Editor#atwhojs-data-module  didn't seem to integrate the autocomplete in pl.org.Any additional changes that you think are needed

**Comment by keshav234156** [comment]
@niravasher

**Comment by cesswairimu** [comment]
Is this in any chance related to https://github.com/publiclab/plots2/pull/8248 if so could we please align and maintain one PR. Thanks

**Comment by cesswairimu** [comment]
@sagarpreet-chadha could you please help with reviewing this. Thanks

**Comment by jywarren** [comment]
Let's see how this does... i think we need to rebase?

**Comment by jywarren** [comment]
OK, i'm seeing:

```
===============[Screenshot]: tmp/screenshots/failures_test_post_note,_then_comment_on_FRESH_note.png
ERROR CommentTest#test_post_note,_then_comment_on_FRESH_note (523.75s)
Minitest::UnexpectedError:         ActionController::RoutingError: No route matches [GET] "/lib/publiclab-editor/data/emoji.js"
            /home/runner/work/plots2/plots2/vendor/bundle/ruby/2.6.0/gems/actionpack-5.2.3/lib/action_dispatch/middleware/debug_exceptions.rb:65:in `call'
```

```
2021-04-15 21:19:01 +0000 Rack app ("GET /lib/publiclab-editor/data/atwho.PLE.js" - (127.0.0.1)): #<ActionController::RoutingError: No route matches [GET] "/lib/publiclab-editor/data/atwho.PLE.js">
======[Screenshot]: tmp/screenshots/failures_test_posting_from_the_editor.png
ERROR PostTest#test_posting_from_the_editor (213.64s)
Minitest::UnexpectedError:         ActionController::RoutingError: No route matches [GET] "/lib/publiclab-editor/node_modules/at.js/dist/css/jquery.atwho.min.css"
```

So we're missing:

* `/lib/publiclab-editor/data/emoji.js`
* `/lib/publiclab-editor/node_modules/at.js/dist/css/jquery.atwho.min.css`

File: app/views/editor/rich.html.erb [hunk 1]
```diff
@@ -15,6 +15,14 @@
 <link href="/lib/publiclab-editor/dist/PublicLab.Editor.css" rel="stylesheet">
 <script src="/lib/publiclab-editor/dist/PublicLab.Editor.js"></script>
 
+<link
+  href="/lib/publiclab-editor/node_modules/at.js/dist/css/jquery.atwho.min.css"
+  rel="stylesheet"
+/>
+<script src="/lib/publiclab-editor/node_modules/at.js/dist/js/jquery.atwho.min.js"></script>
+<script src="/lib/publiclab-editor/data/emoji.js"></script>
```
Comment (jywarren) [inline app/views/editor/rich.html.erb:23]:
```suggestion
<script src="/lib/publiclab-editor/examples/data/emoji.js"></script>
```

File: app/views/editor/rich.html.erb [hunk 2]
```diff
@@ -15,6 +15,14 @@
 <link href="/lib/publiclab-editor/dist/PublicLab.Editor.css" rel="stylesheet">
 <script src="/lib/publiclab-editor/dist/PublicLab.Editor.js"></script>
 
+<link
+  href="/lib/publiclab-editor/node_modules/at.js/dist/css/jquery.atwho.min.css"
+  rel="stylesheet"
+/>
+<script src="/lib/publiclab-editor/node_modules/at.js/dist/js/jquery.atwho.min.js"></script>
```
Comment (jywarren) [inline app/views/editor/rich.html.erb:22]:
```suggestion
<script src="/lib/at.js/dist/js/jquery.atwho.min.js"></script>
```

File: app/views/editor/rich.html.erb [hunk 3]
```diff
@@ -15,6 +15,14 @@
 <link href="/lib/publiclab-editor/dist/PublicLab.Editor.css" rel="stylesheet">
 <script src="/lib/publiclab-editor/dist/PublicLab.Editor.js"></script>
 
+<link
+  href="/lib/publiclab-editor/node_modules/at.js/dist/css/jquery.atwho.min.css"
```
Comment (jywarren) [inline app/views/editor/rich.html.erb:19]:
```suggestion
  href="/lib/at.js/dist/css/jquery.atwho.min.css"
```

File: app/views/editor/rich.html.erb [hunk 4]
```diff
@@ -15,6 +15,14 @@
...
+  rel="stylesheet"
+/>
+<script src="/lib/publiclab-editor/node_modules/at.js/dist/js/jquery.atwho.min.js"></script>
+<script src="/lib/publiclab-editor/data/emoji.js"></script>
+<script src="/lib/publiclab-editor/data/atwho.PLE.js"></script>
```
Comment (jywarren) [inline app/views/editor/rich.html.erb:24]:
```suggestion
<script src="/lib/publiclab-editor/examples/data/atwho.PLE.js"></script>
```

**Comment by jywarren** [comment]
Fixed some CSS and JS paths!!!

**Comment by jywarren** [comment]
Ooohhhhhh! Testing in GitPod!!

**Comment by jywarren** [comment]
Hmmmmm. OK this doesn't work in GitPod. I wonder if the option is getting set correctly? 

https://github.com/publiclab/PublicLab.Editor/pull/48/files

**Comment by jywarren** [comment]
Woohoo!!!

File: app/views/editor/rich.html.erb [hunk 5]
```diff
@@ -483,6 +493,54 @@
...
+        data: Object.keys(emoji).map(function(name){ return {'name': name, 'value': emoji[name]}}),
+        displayTpl: "<li>${value} ${name}</li>",
+        insertTpl: ":${name}:",
+        limit: 100
+    }
+
+    $('.wk-wysiwyg')
+        .atwho(hashtags_config)
+        .atwho(emojis_config)
+        //.atwho(at_config) // off for performance reasons for now
```
Comment (jywarren) [inline app/views/editor/rich.html.erb:541]: @TildaDares here it is! This is where I turned off the username autocompletion. It uses the same API backend as your code. But the front-end implementation may be a little different as it's within a rich text editor!

**Pull request merged**